src/browser-driver-manager.js

```diff
@@ -1,259 +1,259 @@
-        await installChromeDriver(version, verbose);
-        console.log(`Successfully installed ChromeDriver ${version}`);
-
-        return;
-      } else {
-        console.log(
-          `ChromeDriver ${version} not found. Retrying with ChromeDriver ${
-            version - 1
-          }`
-        );
-
-        if (version - 1 <= chromedriverVersion) {
-          await installChromeDriver(version - 1, verbose);
-          console.log(`Successfully installed ChromeDriver ${version - 1}`);
-
-          return;
-        } else {
-          return error(
-            chalk,
-            'Unable to get ChromeDriver version; Something went wrong'
-          );
-        }
-      }
+    const browserTitle = capitalize(browser);
+    let progressMessage = `Downloading ${browserTitle}: `;
+    if (downloadedBytes < totalBytes) {
+      const cursorDisablingString = '\x1B[?25l';
+      progressMessage = `${cursorDisablingString}${progressMessage}`;
+      progressMessage += `${Math.ceil((downloadedBytes * 100) / totalBytes)}%`;
+    } else {
+      const cursorEnablingString = '\r\n\x1B[?25h';
+      progressMessage += `Done!${cursorEnablingString}`;
     }
+    readline.cursorTo(process.stdout, 0);
+    process.stdout.write(progressMessage);
+  };
 
-    return;
+  try {
+    const installedBrowser = await puppeteerInstall({
+      cacheDir,
+      browser,
+      buildId,
+      downloadProgressCallback
+    });
+
+    return installedBrowser;
+  } catch (e) {
+    throw new Error(e);
   }
-  // output version installed from npm
-  else if (['version', 'which'].includes(userArgs[0]) && chromedriver) {
-    try {
-      const chromedriverPkg = require('chromedriver');
+}
 
-      if (userArgs[0] === 'version') {
-        console.log(chromedriverPkg.version);
-      } else {
-        console.log(chromedriverPkg.path);
-      }
+/**
+ * Write the environment file to the BDMCacheDir.
+ * @param {object} data - The data to write.
+ * @param {string} data.chromePath - The path to Chrome.
+ * @param {string} data.chromedriverPath - The path to Chromedriver
+ * @param {string} data.version - The version of Chrome and Chromedriver
+ * @throws {Error} - Environment file must be writable.
+ */
+async function setEnv({ chromePath, chromedriverPath, version }) {
+  console.log('Setting env CHROME/CHROMEDRIVER_TEST_PATH/VERSION');
 
-      return;
-    } catch (err) {
-      return error(chalk, 'chromedriver is not installed');
-    }
+  try {
+    await fsPromises.writeFile(
+      path.resolve(getBDMCacheDir(), '.env'),
+      `CHROME_TEST_PATH="${chromePath}"${os.EOL}CHROMEDRIVER_TEST_PATH="${chromedriverPath}"${os.EOL}VERSION="${version}"${os.EOL}`
+    );
+    console.log('CHROME_TEST_PATH is set in', chromePath);
+    console.log('CHROMEDRIVER_TEST_PATH is set in', chromedriverPath);
+    console.log('VERSION:', version);
+  } catch (e) {
+    throw new Error('Error setting CHROME/CHROMEDRIVER_TEST_PATH/VERSION');
```

Comment:
If user is unable to set the env what should they do in order to proceed forward?